octopusdeploy/schema_runbook.go

```diff
@@ -0,0 +1,182 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func expandRunbook(ctx context.Context, d *schema.ResourceData) *runbooks.Runbook {
+	name := d.Get("name").(string)
+	projectId := d.Get("project_id").(string)
+
+	runbook := runbooks.NewRunbook(name, projectId)
+	runbook.ID = d.Id()
+
+	if v, ok := d.GetOk("description"); ok {
+		runbook.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("runbook_process_id"); ok {
+		runbook.RunbookProcessID = v.(string)
+	}
+
+	if v, ok := d.GetOk("published_runbook_snapshot_id"); ok {
+		runbook.PublishedRunbookSnapshotID = v.(string)
+	}
+
+	if v, ok := d.GetOk("space_id"); ok {
+		runbook.SpaceID = v.(string)
+	}
+
+	if v, ok := d.GetOk("multi_tenancy_mode"); ok {
+		runbook.MultiTenancyMode = core.TenantedDeploymentMode(v.(string))
+	}
+
+	if v, ok := d.GetOk("connectivity_policy"); ok {
+		runbook.ConnectivityPolicy = expandConnectivityPolicy(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("environment_scope"); ok {
+		runbook.EnvironmentScope = v.(string)
+	}
+
+	if v, ok := d.GetOk("environments"); ok {
+		runbook.Environments = getSliceFromTerraformTypeList(v)
+	}
+
+	if v, ok := d.GetOk("default_guided_failure_mode"); ok {
+		runbook.DefaultGuidedFailureMode = v.(string)
+	}
+
+	if v, ok := d.GetOk("retention_policy"); ok {
+		runbook.RunRetentionPolicy = expandRunbookRetentionPolicy(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("force_package_download"); ok {
+		runbook.ForcePackageDownload = v.(bool)
+	}
+
+	return runbook
+}
+
+func getRunbookSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"id": getIDSchema(),
+		"name": {
+			Description:      "The name of the project in Octopus Deploy. This name must be unique.",
+			Required:         true,
+			Type:             schema.TypeString,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+		},
+		"description": {
+			Computed:    true,
+			Description: "The description of this runbook.",
+			Optional:    true,
+			Type:        schema.TypeString,
+		},
+		"project_id": {
+			Description: "The project that this runbook belongs to.",
+			Required:    true,
+			Type:        schema.TypeString,
+		},
+		"runbook_process_id": {
+			Description: "The runbook process ID.",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"published_runbook_snapshot_id": {
+			Description: "The published snapshot ID.",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"space_id": {
+			Computed:         true,
+			Description:      "The space ID associated with this project.",
+			Optional:         true,
+			Type:             schema.TypeString,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+		},
+		"multi_tenancy_mode": getTenantedDeploymentSchema(),
+		"connectivity_policy": {
+			Computed: true,
+			Elem:     &schema.Resource{Schema: getConnectivityPolicySchema()},
+			MaxItems: 1,
```

Comment:
@jbristowe  I know this is the way we did it in `resource_project` but why is this a list?